4-Intermediario/1-Funcoes.md

```diff
@@ -254,6 +256,41 @@ $exemplo = function () use ($mensagem)
 };
 $exemplo(); // Resultado: He4rtDevs
 
+```
+### Callbacks
+Bom com alguns conceitos explicados, pode ter ficado com um nó na cabeça, mas afinal o que é um **CALLBACK** ?
+
+Resumidamente, é um nome para uma função, que vai ser passada como paramêtro dentro de outra função, calma... Vamos aos exemplos.
+
+Primeiramente, vamos tentar algo mais simples, por exemplo podemos passar uma função da propria linguagem para execução de uma tarefa:
+```php
+    array_map('trim',['  oi tudo bem ? ', ' Como vai você?', 'Beleza ?  ']); // Resultado: ['oi tudo bem?', 'Como vai você ?', 'Beleza ?']
+```
+
+Bom, explicando de uma maneira, mais declarativa, podemos interpretar, como quando sua mãe fala explicitamente para você:
+    - "Filho, por favor lave a louça para mim, enquanto vou ao mercado. Se fizer lhe dou um doce. Se não, vou te colocar de castigo."
+
+```php
+    function pedirParaLavarLouca($filho, $callback){
+        $resultado = lavouLouca();
+
+        if($resultado === true){
```

Comment:
Então, a expressão "x === true" é algo que podemos entender de fato, mas precisa forçar tanto esse true? 
Eu vejo que não tem necessidade, porém se tratando de um 4noobs dá pra entender o motivo.
Caso você achar válido, refatore. 

Coloca o espaçamento da PSR e boa!

Reply:
Sim, esse === true, é de proposito, justamente porque estamos falando de um conteúdo pra iniciantes, e trollei realmente no espaçamento é valido a correção.